src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -170,8 +170,8 @@ public Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operation>> allOpe
 
         public virtual Resource GetResource() => this;
 
-        //private string? _defaultName;
-        protected override string DefaultName => ResourceName;
+        // name after `{ResourceName}Resource`, unless the `ResourceName` already ends with `Resource`
+        protected override string DefaultName => Configuration.MgmtConfiguration.NoResourceSuffix.Contains(ResourceName) ? ResourceName : ResourceName.AddResourceSuffixToResourceName();
```

Comment:
I am confused by this new configuration and think it might be redundant.
Which scenario is this new configuration for? Could you please explain how to use this new configuration in detail?
For instance, the word `GenericResource` is in this list, in this case, you will get the result `GenericResource` (the first branch is executed). If `GenericResource` is not in the list, you will still get the result `GenericResource` (the second branch is executed). Therefore which scenario this new configuration is designed for?

Reply:
Also this new configuration does not work with the name of a resource data (because I did not see any changes in the implementation of resource data), therefore the "trimming resource out of resource data's name" is still achieved by the `rename-model` directive. Is this statement correct?

Reply:
That's correct. The latest discussion is that service team still need to `rename-model` to remove the `Resource` from their swagger definition. So:
- If you have `Tenant` in swagger, then you'll get `TenantData`, `TenantResource`, `TenantCollection`.
- If you have `PrivateLinkResource` in swagger, then you'll get `PrivateLinkResourceData`, `PrivateLinkResource`, and `PrivateLinkResourceCollection`.

The new configuration is for `ResourceGroup`, `ResourceLink` and `ResourceProvider` (previously `Provider`). When we do not want to append `Resource` suffix. So if I add `no-resource-suffix: ResourceGroup`, I'll get `ResourceGroupData`, `ResourceGroup`, and `ResourceGroupCollection`.

Reply:
In the latest discussion, we agree to keep `no-resource-suffix` for a while. If no usage, I'll move it from `autorest.csharp`.

Reply:
I see the purpose - and it is useful in some cases. But I suggest we change the key of this configuration, instead of using the name of the resource as keys, we should use the request path as the key of this configuration.
The name of a resource could be generated programmatically, and it might change if new swaggers are added to the tag - but the request path of a resource is unique.